datashader/core.py

```diff
@@ -264,7 +264,11 @@ def raster(self,
         w = int(np.ceil(self.plot_width * width_ratio))
         h = int(np.ceil(self.plot_height * height_ratio))
 
-        data = resample_2d(source.values[0].astype(np.float32),
+        rmin, cmin = get_indices(xmin, ymin, source.x.values, source.y.values, res)
+        rmax, cmax = get_indices(xmax, ymax, source.x.values, source.y.values, res)
```

Comment:
Please do push the fix; @gbrener is away today, and he can make a new PR tomorrow if this wasn't the correct approach.  Better for us to be able to move on and merge this and your other PR.